x-pack/plugins/enterprise_search/public/applications/__mocks__/kea.mock.ts

```diff
@@ -4,21 +4,46 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+/**
+ * NOTE: These variable names MUST start with 'mock*' in order for
+ * Jest to accept its use within a jest.mock()
+ */
+import { mockHttpValues } from './http_logic.mock';
+import { mockFlashMessagesValues, mockFlashMessagesActions } from './flash_messages_logic.mock';
+
+export const mockAllValues = {
+  ...mockHttpValues,
+  ...mockFlashMessagesValues,
+};
+export const mockAllActions = {
+  ...mockFlashMessagesActions,
+};
+
+/**
+ * Import this file directly to mock useValues with a set of default values for all shared logic files.
+ * Example usage:
+ *
+ * import '../../../__mocks__/kea'; // Must come before kea's import, adjust relative path as needed
+ */
 jest.mock('kea', () => ({
   ...(jest.requireActual('kea') as object),
-  useValues: jest.fn(() => ({})),
-  useActions: jest.fn(() => ({})),
+  useValues: jest.fn(() => ({ ...mockAllValues })),
+  useActions: jest.fn(() => ({ ...mockAllActions })),
 }));
 
 /**
+ * Call this function to override a specific set of Kea values while retaining all other defaults
  * Example usage within a component test:
  *
- * import '../../../__mocks__/kea'; // Must come before kea's import, adjust relative path as needed
- *
- * import { useActions, useValues } from 'kea';
+ * import '../../../__mocks__/kea';
+ * import { setMockValues } from ''../../../__mocks__';
  *
  * it('some test', () => {
- *   (useValues as jest.Mock).mockImplementationOnce(() => ({ someValue: 'hello' }));
- *   (useActions as jest.Mock).mockImplementationOnce(() => ({ someAction: () => 'world' }));
+ *   setMockValues({ someValue: 'hello' });
  * });
  */
+import { useValues } from 'kea';
+
+export const setMockValues = (values: object) => {
+  (useValues as jest.Mock).mockImplementation(() => ({ ...mockAllValues, ...values }));
+};
```

Comment:
Oh, I totally forgot to give you credit for this @scottybollinger! I stole this idea from your awesome overview mocks 😁